File: vendas-ui/src/app/models/item.ts
import { Produto } from "./produto";
import { Venda } from "./venda";

export class Item {
  id: string = '';
  venda?: Venda;
  produto: Produto;
  quantidade: number = 0; 

  public toDTO():ItemDTO{
      return new ItemDTO(this);
  }
}

export class ItemDTO {
  produtoId: string = '';
  quantidade: number = 0;

  constructor(item: Item) {
    this.produtoId = item.produto.id;
    this.quantidade = item.quantidade;
  }
}

File: api-vendas/src/main/kotlin/br/com/juniorstreichan/apivendas/domain/Item.kt
package br.com.juniorstreichan.apivendas.domain

import com.fasterxml.jackson.annotation.JsonIgnore
import org.jetbrains.annotations.NotNull
import javax.persistence.*

@Entity
data class Item(

        @Id
        @GeneratedValue(strategy = GenerationType.IDENTITY)
        val id: Int = 0,


        @JsonIgnore
        @ManyToOne
        val venda: Venda = Venda(),

        @NotNull
        @ManyToOne
        val produto: Produto = Produto(),

        private val descricao: String = produto.descricao,

        @NotNull
        val quantidade: Int = 0

){
        fun getDescricao():String = descricao
}